qiskit/providers/ibmq/ibmqfactory.py

```diff
@@ -102,7 +136,27 @@ def enable_account(
                            'account.')
             return None
 
-        return providers[0]
+        # The provider for the default open access project.
+        default_provider = providers[0]
+
+        # If any `hub`, `group`, or `project` is specified, make sure all parameters are set.
+        if any([hub, group, project]) and not all([hub, group, project]):
+            raise IBMQAccountValueError('The hub, group, project parameters must all be '
+                                        'specified when using a default provider: '
+                                        'hub = "{}", group = "{}", project = "{}"'
+                                        .format(hub, group, project))
+
+        # If `hub`, `group`, `project` are all specified, attempt to get the corresponding provider.
+        if all([hub, group, project]):
+            try:
+                default_provider = self.get_provider(hub=hub, group=group, project=project)
+            except IBMQProviderError as ex:
+                raise IBMQAccountError('The default provider (hub/group/project) '
+                                       'to use could not be found:'
+                                       'hub = "{}", group = "{}", project = "{}"'
+                                       .format(hub, group, project)) from ex
```

Comment:
Why convert `IBMQProviderError`, which is already a subclass of `IBMQAccountError`, to a new `IBMQAccountError`?

Reply:
I was re-raising in order to add a bit more context to the exception message. But, you're right, it might not be necessary, since the one provided by `IBMQProviderError` should be straightforward enough when enabling an account.